src/Features/Core/Portable/Completion/CompletionRules.cs

```diff
@@ -20,20 +21,22 @@ internal class CompletionRules
         private readonly object _gate = new object();
         private readonly AbstractCompletionService _completionService;
         private readonly Dictionary<string, PatternMatcher> _patternMatcherMap = new Dictionary<string, PatternMatcher>();
+        private readonly Dictionary<string, PatternMatcher> _enUSCulturePatternMatcherMap = new Dictionary<string, PatternMatcher>();
```

Comment:
Yes that could a problem, but regardless of what lookup you persist you’ll run into this problem.

If we tackle this, we should do as a separate issue.

From: Ravi Chande [mailto:notifications@github.com]
Sent: Tuesday, October 27, 2015 9:35 AM
To: dotnet/roslyn <roslyn@noreply.github.com>
Cc: David Kean <David.Kean@microsoft.com>
Subject: Re: [roslyn] When filtering completion in the Turkish locale, attempt matches in the current culture as well as the invariant culture (#6266)


In src/Features/Core/Portable/Completion/CompletionRules.cs<https://na01.safelinks.protection.outlook.com/?url=https%3a%2f%2fgithub.com%2fdotnet%2froslyn%2fpull%2f6266%23discussion_r43147794&data=01%7c01%7cDavid.Kean%40microsoft.com%7c4903c28800ea45b867ca08d2deec985e%7c72f988bf86f141af91ab2d7cd011db47%7c1&sdata=z6KDjNWhr5sV4Sd65SevMKA2GxgJ6UiiXM%2fp%2bg0CxG8%3d>:

> @@ -20,20 +21,22 @@ internal class CompletionRules

>          private readonly object _gate = new object();

>          private readonly AbstractCompletionService _completionService;

>          private readonly Dictionary<string, PatternMatcher> _patternMatcherMap = new Dictionary<string, PatternMatcher>();

> +        private readonly Dictionary<string, PatternMatcher> _enUSCulturePatternMatcherMap = new Dictionary<string, PatternMatcher>();

@davkean<https://na01.safelinks.protection.outlook.com/?url=https%3a%2f%2fgithub.com%2fdavkean&data=01%7c01%7cDavid.Kean%40microsoft.com%7c4903c28800ea45b867ca08d2deec985e%7c72f988bf86f141af91ab2d7cd011db47%7c1&sdata=AO4MXg0DeXPcQexEfpUO%2bCupY2l58I99GAUQ%2bpH%2fybg%3d> Does doing ordinal lookups here mean we could have a bug where someone changes their culture and we start returning cached matches against their previous culture?

—
Reply to this email directly or view it on GitHub<https://na01.safelinks.protection.outlook.com/?url=https%3a%2f%2fgithub.com%2fdotnet%2froslyn%2fpull%2f6266%2ffiles%23r43147794&data=01%7c01%7cDavid.Kean%40microsoft.com%7c4903c28800ea45b867ca08d2deec985e%7c72f988bf86f141af91ab2d7cd011db47%7c1&sdata=ys9sngrvshy1QB%2fnv6Bata43HOxBNJGJ5ZzRAPcFx84%3d>.